### ISSUE TITLE
Let's Encrypt certificate is not renewed automatically

### ISSUE THREAD

**Issue by mapreri** [globaleaks/GlobaLeaks#4132]
### What version of GlobaLeaks are you using?

4.15.6

### What operating system(s) are you seeing the problem on?

Linux

### Describe the issue

The certificate obtained via Let's Encrypt using the included LE client is never renewed, despite the "Auto-renewal: Enabled" flag.

Not sure what might be going on, this could be a configuration issue on my side.

**Comment by evilaliv3** [comment]
Hello @mapreri 

Would you please check that both port 80 and 443 are open publicly and that are both directly handled by GlobaLeaks without any intermediate proxy?

I suspect you may have either port 80 closed or implementing a redirect to port 443.

If you could pass to me the address of your server i could verify the exact issue.

**Comment by mapreri** [comment]
They are open and handled by the standard globaleaks iptables rules.

You can look at https://leaks.bluebirdind.com/ for example.

I must add that I don't think renewal ever worked, every 3 months I found
myself disabling and re-enabling LE to obtain a new cert (last time this
morning)

**Comment by evilaliv3** [comment]
Thank you @mapreri , this is actually quite strange. If you could share to me the access log i will try to see which is the reason.  Near to the expiration the application starts requesting renewal with a request every day; Do you have some firewalls rules that prevent outgoing connections?

P.s.: I acknowledge that you have removed the "Powered by GlobaLeaks" attribution clause; this is actually in violation of the software license: https://github.com/globaleaks/GlobaLeaks/blob/main/LICENSE
It is not a problem as long that you restore it timely by before 30 days since this notification.
Thank you for your understanding.

**Comment by mapreri** [comment]
> Thank you @mapreri , this is actually quite strange. If you could share to me the access log i will try to see which is the reason. Near to the expiration the application starts requesting renewal with a request every day; Do you have some firewalls rules that prevent outgoing connections?

I don't have any firewall rules limiting outgoing connections.

What's the best way to share the access.log privately to you?

> 
> P.s.: I acknowledge that you have removed the "Powered by GlobaLeaks" attribution clause; this is actually in violation of the software license: https://github.com/globaleaks/GlobaLeaks/blob/main/LICENSE It is not a problem as long that you restore it timely by before 30 days since this notification. Thank you for your understanding.

AFAIK it's not a violation of the AGPL as long as the code running is completely unmodified from what I originally obtained by the licensor (which it is, in this case).  Nevertheless, I reckon this customer is kinda ill-advised, so I'm going behind his back and reinstating the line :stuck_out_tongue_winking_eye:  - I am a fairly active FOSS sustainer after all heh

**Comment by mapreri** [comment]
> AFAIK it's not a violation of the AGPL as long as the code running is completely unmodified from what I originally obtained by the licensor (which it is, in this case). Nevertheless, I reckon this customer is kinda ill-advised, so I'm going behind his back and reinstating the line 😜 - I am a fairly active FOSS sustainer after all heh

I see now that it's actually an addendum to the AGPL that you did.  That is fine, however I recommend you add a note in the README mentioning that you have additional terms to the AGPL, as I know that nobody reads the full LICENSE document after they see a standard FOSS license (I already read nearly all of them more than once, I can do without reading them all over once more…)

**Comment by evilaliv3** [comment]
Thank you @mapreri  for your feedback.

Actually we were listing such a notice in the README.md but we removed it considering the license was enough.

I just re-added them with commit: https://github.com/globaleaks/GlobaLeaks/commit/bdef24b2fc260bedb855e93c60c6f1e135c71ced

**Comment by mapreri** [comment]
No, I haven't found anything relevant with a quick grep of the logs tbh.  What should I be looking for?

Else, I'm fine sending them to you if you can provide a... email address and a gpg key to encrypt to I suppose?

**Comment by evilaliv3** [comment]
Sure,

https://www.evilaliv3.org/0x67A0F187.asc

**Comment by mapreri** [comment]
I still seem to have this problem, and today a bunch of certificates expired.  currently running 5.0.34 when the certs expired.

**Comment by evilaliv3** [comment]
Do you have port 80 open on your firewall?

The letsencrypt protocol requires globaleaks to answer on that port when renewing certificates

**Comment by mapreri** [comment]
yes, 80 it's open.  Besides the LE certificates are correctly issues just fine when I click the button, they are just not auto-renewed (plus there is no manual renew option, so I need to reset the SSL configuration and re-issue a new cert).

**Comment by evilaliv3** [comment]
I see, thank you for this update.

Do you have a link to your setup that you could share?

We will try to see in the future if we could at least implement manual renewal to improve this situation.

**Comment by mapreri** [comment]
You can check this (encrypted address, I'd rather not advertise it):

```
-----BEGIN PGP MESSAGE-----

hQIMA7sf45vpbp73ARAAkzEmXRR4W6l5Nr/S75CKCld41QLv35Qx8k2hE0cKc5VD
lrLe2avgblEreZhE6UrCC6WdfMkVViDId5I1Lgnpbs9RwPwErVLi0eokZjI5ywQ8
pk/naQwpb56+oVMOBINmU54IVII0rNpCd0rdkGojpeneMXiB3lTRqSRIfTGhtLEY
RCWP0IL/ukX0tlNGaZkp+ZxLBoUuD+9j1mCapEq7Rbw0fLk7MD2bMIFsYESJbtlk
LcNhTcmO7LB2CmA4BKOx2ACUEn+ExE4U7MDJS8jp9OyoSqTWVc8lbD6XX/kUNqoj
RROUKKKPykQO1Vm+6cOtnohyGIBDgQGQWrcC5lplyCPPNYjH2XaBCuwcdYXUFyKg
UT6JH8QFQY2xJoJpQDecidoNpwI5TzvUsk1HHjwGrgBhP1P/41gXwU8O51l5L287
dVWh9R/IP3SRd4ouR5vSU/Qzff1KXGRxJSM+iTPgEfotZFZ/KTAGLxXd0aZCgqrm
a7iTJVU3xXbv2GTc5Vzt7nkBrdrX1haJ/96ASlK3yB9Ainfr/114hTeMZXf2V8QE
xhZJC3+ViirTB0SyS8q//ob/aXG7C6zUFVgVP/USv/GTK+M0iCjD9LZ/GhPnjqK/
b3RWUonjbPBZCw6J+1UFzth19zVE0jJQ2Cky0T8ObZLta+xKNoobVYOE871XYXPS
WgEjpfnSHLcdSOqLK68a5Km/nRVF75l08T4oL2TTl/vRTYVg4jkWJ4zrgqYulMy8
Tpyo7ymf5yMjZZ1CSTgPFF37yRwsqbTt24y0W7Oxw1uBd0dEdK2Qy4xkQw==
=NIH9
-----END PGP MESSAGE-----
```
